CapstoneProject/Assets/Scripts/Core/BehaviorTree/CoreNodeTypes/CompositeNode.cs

```diff
@@ -4,5 +4,12 @@
 
 public abstract class CompositeNode : Node
 {
-    public List<Node> children = new List<Node>();
+    [HideInInspector] public List<Node> children = new List<Node>();
+
+    public override Node Clone()
+    {
+        CompositeNode node = Instantiate(this);
+        node.children = children.ConvertAll(child => child.Clone());
+        return node;
```

Comment:
Maybe I'm not understanding correctly the functionality here, but this function seems like it is recursively going to keep cloning children? Does this not create too many children though as each child would keep making more instances?